kategory-docs/docs/typeclasses/applicative/README.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: docs
+title: Applicative
+permalink: /docs/typeclasses/applicative/
+---
+
+## Functor
+
+The `Applicative` typeclass abstracts the ability to lift values and apply functions over the computational context of a type constructor.
+Examples of type constructors that can implement instances of the Applicative typeclass include `Option`, `NonEmptyList`,
+`List` and many other datatypes that include a `pure` and either `ap` function. `ap` may be derived for monadic types that include a `Monad` instance via `flatMap`.
+
+### Applicative Builder examples
+
+Often times we find ourselves in situations where we need to compute multiple independent values resulting from operations that do not depend on each other.
+
+In the following example we will define 3 invocations that may as well be remote or local services each one of them returning different results in the same computational context of `Option`
+
+```kotlin:ank
+import kategory.*
+
+fun profileService(): Option<String> = Option("Alfredo Lambda")
+fun phoneService(): Option<Int> = Option(55555555)
+fun addressService(): Option<List<String>> = Option(listOf("1 Main Street", "11130", "NYC"))
+```
+
+This more or less illustrate the common use case of performing several independent operations where we need to get all the results together
+
+Kategory features an [Applicative Builder](/docs/patterns/applicativebuilder) that allows you to easily combine all the independent operations into one result.
+
+```kotlin:ank
+data class Profile(val name: String, val phone: Int, val address: List<String>)
+
+val r: Option<Tuple3<String, Int, List<String>>> = Option.tupled(profileService(), phoneService(), addressService())
+r.map { Profile(it.a, it.b, it.c) } 
+```
+
+The Applicative Builder also provides a `map` operations that is able to abstract over arity in the same way as `tupled`
+
+```kotlin:ank
+Option.map(profileService(), phoneService(), addressService(), { name, phone, address ->
+  Profile(name, phone, address)
+})
+```
+
+### Main Combinators
+
+#### pure
+
+Lifts a value into the computational context of a type constructor
+
+`fun <A> pure(a: A): HK<F, A>`
+
+```kotlin:ank
+Option.pure(1) // Option(1)
+```
+
+#### ap
+
+Apply a function inside the type constructor's context
+
+`fun <A, B> ap(fa: HK<F, A>, ff: HK<F, (A) -> B>): HK<F, B>`
```

Comment:
This is the most generic example of the API. We should display some implementation examples below, i.e. the same for Option or Either. Not the implementation, just the signature with the generics pre-applied.

Reply:
There are examples below of usage. Adding a concrete form using for example `OptionKind` it's going to generate more confusion since we will have to explain here what kinds are. This can be better addressed in the derivation section or a section where we explain how we use higher kinds and how we have emulated them.